File: safety_monitor/frontend/src/api/recording.ts
import request from './request'

export interface Recording {
  id: number
  camera_id: number
  camera_name?: string
  file_path: string
  start_time: string
  end_time?: string
  duration?: number
  file_size?: number
  status: string
  created_at: string
}

export const recordingApi = {
  list: (params?: {
    camera_id?: number
    start_date?: string
    end_date?: string
    status?: string
    skip?: number
    limit?: number
  }) => {
    return request.get('/recordings', { params })
  },

  get: (id: number) => {
    return request.get(`/recordings/${id}`)
  },

  create: (data: Partial<Recording>) => {
    return request.post('/recordings', data)
  },

  update: (id: number, data: Partial<Recording>) => {
    return request.put(`/recordings/${id}`, data)
  },

  delete: (id: number) => {
    return request.delete(`/recordings/${id}`)
  },

  getByCamera: (cameraId: number, date?: string) => {
    return request.get(`/recordings/cameras/${cameraId}/recordings`, {
      params: date ? { date } : {}
    })
  }
}

File: safety_monitor/backend/app/schemas/recording.py
from pydantic import BaseModel
from typing import Optional
from datetime import datetime


class RecordingBase(BaseModel):
    camera_id: int
    file_path: str
    start_time: datetime
    end_time: Optional[datetime] = None
    duration: Optional[int] = None
    file_size: Optional[int] = None
    status: str = "recording"


class RecordingCreate(RecordingBase):
    pass


class RecordingUpdate(BaseModel):
    end_time: Optional[datetime] = None
    duration: Optional[int] = None
    file_size: Optional[int] = None
    status: Optional[str] = None


class RecordingResponse(RecordingBase):
    id: int
    created_at: datetime

    class Config:
        from_attributes = True


class RecordingListResponse(BaseModel):
    id: int
    camera_id: int
    camera_name: Optional[str] = None
    file_path: str
    start_time: datetime
    end_time: Optional[datetime] = None
    duration: Optional[int] = None
    file_size: Optional[int] = None
    status: str
    created_at: datetime

    class Config:
        from_attributes = True
